src/services/discord.ts

```diff
@@ -77,11 +93,47 @@ export const getToken = async (grantCode: string) => {
  * and throw an error if no response was received during that delay.
  */
 export const fetchDiscordUser = async (discordApiToken: DiscordToken) => {
-  const userRequest = await discordFetcher.get<DiscordAuthorizationData>(`${env.discord.oauthUrl}/@me`, {
+  const userRequest = await discordFetcher.get<DiscordAuthorizationData>(`${env.discord.apiUrl}/oauth2/@me`, {
     headers: {
       // Authenticate to discord api with Bearer user token
       Authorization: `Bearer ${discordApiToken.access_token}`,
     },
   });
   return userRequest.data.user;
 };
+
+export const createDiscordChannel = async (requestBody: DiscordCreateChannelRequest) => {
+  const request = await bot.post<DiscordChannel>(`guilds/${env.discord.server}/channels`, requestBody);
+
+  return request.data;
+};
+
+/**
+ * Create a discord role
+ */
+export const createDiscordRole = async (requestBody: DiscordCreateRoleRequest) => {
+  const request = await bot.post<DiscordRole>(`guilds/${env.discord.server}/roles`, requestBody);
+
+  return request.data;
+};
+
+export const findDiscordRoleById = async (id: Snowflake) => {
+  const request = await bot.get<DiscordRole[]>(`guilds/${env.discord.server}/roles`);
+
+  return request.data.find((role) => role.id === id);
+};
+
+export const findDiscordEveryoneRole = async () => {
+  const request = await bot.get<DiscordRole[]>(`guilds/${env.discord.server}/roles`);
+
+  return request.data.find((role) => role.name === '@everyone');
+};
```

Comment:
Au lieu de faire le call de l'api discord pour chaque tournoi, ce serait mieux de tous les garder et de prendre celui qui correspond au niveau de la boucle for